ra2ce/graph/graph_files/graph_files_collection.py

```diff
@@ -0,0 +1,185 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from pathlib import Path
+
+from geopandas import GeoDataFrame
+from networkx import MultiGraph
+
+from ra2ce.graph.graph_files.graph_file import GraphFile
+from ra2ce.graph.graph_files.graph_files_protocol import GraphFileProtocol
+from ra2ce.graph.graph_files.network_file import NetworkFile
+
+
+@dataclass
+class GraphFilesCollection:
+    base_graph: GraphFile = field(
+        default_factory=lambda: GraphFile(name="base_graph.p")
+    )
+    base_graph_hazard: GraphFile = field(
+        default_factory=lambda: GraphFile(name="base_graph_hazard.p")
+    )
+    origins_destinations_graph: GraphFile = field(
+        default_factory=lambda: GraphFile(name="origins_destinations_graph.p")
+    )
+    origins_destinations_graph_hazard: GraphFile = field(
+        default_factory=lambda: GraphFile(name="origins_destinations_graph_hazard.p")
+    )
+    base_network: NetworkFile = field(
+        default_factory=lambda: NetworkFile(name="base_network.feather")
+    )
+    base_network_hazard: NetworkFile = field(
+        default_factory=lambda: NetworkFile(name="base_network_hazard.feather")
+    )
+
+    @property
+    def _graph_collection(self) -> list[GraphFile | NetworkFile]:
+        return [
+            self.base_graph,
+            self.base_graph_hazard,
+            self.origins_destinations_graph,
+            self.origins_destinations_graph_hazard,
+            self.base_network,
+            self.base_network_hazard,
+        ]
+
+    def has_graphs(self) -> bool:
+        """
+        Tests if any of the types has a graph already read
+
+        Returns:
+            bool: True if any graph already read
+        """
+        return any(
+            iter([x for x in self._graph_collection if x.graph is not None]),
+        )
```

Comment:
This is conceptually wrong, when you put a list comprehension between brackets `[ ]` you are in fact running through the whole collection and converting it into a list ( cost `n`), `iter` converts it now into an iterable for then checking whether there are elements with `any`.

When we use `any` ideally you want to finish as soon as possible to save up computational costs (best case scenario, first iteration thus cost = `1`, worst case scenario the whole collection, cost = `n`). Therefore you don't need either `iter` nor the cast to list `[ ]`.
```suggestion
        return any(x for x in self._graph_collection if x.graph is not None)
```

Or something slightly more elegant by using `filter`:
```suggestion
        def has_graph(gf: GraphFileProtocol) -> bool:
            return gf.graph is not None

        return any(filter(has_graph, self._graph_collection))
```

Reply:
Nice